suite/tests/CMakeLists.txt

```diff
@@ -3737,6 +3737,28 @@ if (BUILD_CLIENTS)
     set(tool.drcacheoff.windows-split_postcmd3
       "firstglob@${drcachesim_path}@-indir@${dir_prefix}.*.dir/raw/window.0002@-simulator_type@basic_counts")
 
+    # Test L0_filter_until_instrs
+    torunonly_drcacheoff(L0_filter_until_instrs-windows-split ${ci_shared_app}
+        "-trace_after_instrs 10K -L0_filter_until_instrs 10K -trace_for_instrs 10K -retrace_every_instrs 10K"
+      "@-simulator_type@basic_counts" "")
+    # The first postcmd did window #0.  There should be at least 2 more windows.
+    set(tool.drcacheoff.L0_filter_until_instrs-windows-split_postcmd2
+      "firstglob@${drcachesim_path}@-indir@${dir_prefix}.*.dir/raw/window.0001@-simulator_type@basic_counts")
+    set(tool.drcacheoff.L0_filter_until_instrs-windows-split_postcmd3
+      "firstglob@${drcachesim_path}@-indir@${dir_prefix}.*.dir/raw/window.0002@-simulator_type@basic_counts")
+
+    torunonly_drcacheoff(L0_filter_until_instrs-exit-after ${ci_shared_app}
+        "-trace_after_instrs 10K -L0_filter_until_instrs 10K -exit_after_tracing 10K"
+      "@-simulator_type@basic_counts" "")
+
+    torunonly_drcacheoff(L0_filter_until_instrs-max_refs ${ci_shared_app}
+        "-trace_after_instrs 10K -L0_filter_until_instrs 10K -max_global_trace_refs 10K"
+      "@-simulator_type@basic_counts" "")
+
+    torunonly_drcacheoff(L0_filter_until_instrs-max_trace_size ${ci_shared_app}
+        "-trace_after_instrs 10K -L0_filter_until_instrs 10K -max_trace_size 10K"
+      "@-simulator_type@basic_counts" "")
+
```

Comment:
Could you also add a test with -align_endpoints (which also uses a special drbbdup mode).

Reply:
I mean -align_endpoints without -trace_after_instrs.

I assume you'll remove -trace_after_instrs in at least some of these tests after that bug is fixed right?

Reply:
> Could you also add a test with -align_endpoints (which also uses a special drbbdup mode).

I haven't used this option before. Let me take a look.

> I assume you'll remove -trace_after_instrs in at least some of these tests after that bug is fixed right?

Yes, we should have tests without -trace_after instrs too.

Reply:
Okay, so the idea behind this option is to attach to all threads using the bbdup NOP mode, then switch to tracing (using event_post_attach) and then do the opposite i.e., switch to NOP and then detach.

Are burst_threads/burst_threads_counts the only tests that use -align_endpoints? It doesn't look like this test has enough footprint to cross the filter threshold.

Reply:
> Okay, so the idea behind this option is to attach to all threads using the bbdup NOP mode, then switch to tracing (using event_post_attach) and then do the opposite i.e., switch to NOP and then detach.

Yes.

> Are burst_threads/burst_threads_counts the only tests that use -align_endpoints? It doesn't look like this test has enough footprint to cross the filter threshold.

Not sure I understand. -align_endpoints could be added to any new test, and also we could lower the filter thresholds if the test has too small footprint.

Reply:
I was just trying to understand if I needed to add a new test. Looks like I was mistaken about the footprint part. I was not seeing any L0_filter related message but it was an issue with the test.

I need to get multi-threading working correctly before I can add an -align_endpoints test.

Reply:
We can enable multi-threading support in a separate PR.

Reply:
This seems pretty fundamental: what is missing for thread support?  Is there an up-front check or is it documented by the option so users will know?  Supporting only a single thread makes the feature unusable for most of our use cases.  Do you plan to address that quickly?  Is it going to change how the feature works: what is going to change?